File: CryptoBridge-node/src/domain/entities/Exchange.js
class Exchange {
    constructor(name, symbols = []) {
        if (!name || typeof name !== 'string' || name.trim() === '') {
            throw new Error('Exchange name must be a non-empty string');
        }

        this.name = name;
        this.symbols = Array.isArray(symbols) ? symbols : [];
    }

    hasSymbol(symbol) {
        return this.symbols.includes(symbol);
    }
}

module.exports = Exchange;

File: CryptoBridge-python/src/domain/entities/exchange.py
from dataclasses import dataclass
from typing import List


@dataclass
class Exchange:
    name: str
    symbols: List[str]

    def __post_init__(self):
        if not self.name or not isinstance(self.name, str):
            raise ValueError("Exchange name must be a non-empty string")

        if not isinstance(self.symbols, list):
            self.symbols = []

    def has_symbol(self, symbol: str) -> bool:
        return symbol in self.symbols